lib/src/markdown.module.ts

```diff
@@ -25,18 +23,19 @@ const sharedDeclarations = [
   MarkdownPipe,
 ];
 
+/**
+ * @deprecated use provideMarkdown instead
+ * */
 @NgModule({
-  imports: [CommonModule],
+  imports: [CommonModule, ...sharedDeclarations],
   exports: sharedDeclarations,
-  declarations: sharedDeclarations,
 })
 export class MarkdownModule {
   static forRoot(markdownModuleConfig?: MarkdownModuleConfig): ModuleWithProviders<MarkdownModule> {
     return {
       ngModule: MarkdownModule,
       providers: [
         MarkdownService,
-        markdownModuleConfig && markdownModuleConfig.loader || [],
         markdownModuleConfig && markdownModuleConfig.clipboardOptions || [],
         markdownModuleConfig && markdownModuleConfig.markedOptions || [],
         {
```

Comment:
I believe you should be able to replace what's in the `providers` array with the new `provideMarkdown` function, so that this isn't duplicated.